fly.io/cdn-2024-01-26/default.vcl

```diff
@@ -0,0 +1,117 @@
+# - https://github.com/magento/magento2/blob/03621bbcd75cbac4ffa8266a51aa2606980f4830/app/code/Magento/PageCache/etc/varnish6.vcl
+# - https://abhishekjakhotiya.medium.com/magento-internals-cache-purging-and-cache-tags-bf7772e60797
+
+backend default {
+  .host = "top1.nearest.of.changelog-2024-01-12.internal";
+  .host_header = "changelog-2024-01-12.fly.dev";
+  .port = "4000";
+  .first_byte_timeout = 5s;
+  .probe = {
+    .url = "/health";
+    .timeout = 2s;
+    .interval = 5s;
+    .window = 10;
+    .threshold = 5;
+  }
+}
+
+# https://varnish-cache.org/docs/7.4/users-guide/vcl-grace.html
+# https://docs.varnish-software.com/tutorials/object-lifetime/
+# https://www.varnish-software.com/developers/tutorials/http-caching-basics/
+# https://blog.markvincze.com/how-to-gracefully-fall-back-to-cache-on-5xx-responses-with-varnish/
+sub vcl_backend_response {
+  # Objects within ttl are considered fresh.
+  set beresp.ttl = 60s;
+
+  # Objects within grace are considered stale.
+  # Serve stale content while refreshing in the background.
+  # 🤔 QUESTION: should we vary this based on backend health?
+  set beresp.grace = 24h;
+
+  if (beresp.status >= 500) {
+    # Don't cache a 5xx response
+    set beresp.uncacheable = true;
+
+    # If is_bgfetch is true, it means that we've found and returned the cached
+    # object to the client, and triggered an asynchoronus background update. In
+    # that case, since backend returned a 5xx, we have to abandon, otherwise
+    # the previously cached object would be erased from the cache (even if we
+    # set uncacheable to true).
+    if (bereq.is_bgfetch) {
+      return (abandon);
+    }
+  }
+
+  # 🤔 QUESTION: Should we configure beresp.keep?
+}
+
+# NOTE: vcl_recv is called at the beginning of a request, after the complete
+# request has been received and parsed. Its purpose is to decide whether or not
+# to serve the request, how to do it, and, if applicable, which backend to use.
+sub vcl_recv {
+  # https://varnish-cache.org/docs/7.4/users-guide/purging.html
+  if (req.method == "PURGE") {
+    return (purge);
+  }
+
+  # Implement a Varnish health-check
+  if (req.method == "GET" && req.url == "/varnish_status") {
+    return(synth(204));
+  }
+}
+
+# https://gist.github.com/leotsem/1246511/824cb9027a0a65d717c83e678850021dad84688d#file-default-vcl-pl
+# https://varnish-cache.org/docs/7.4/reference/vcl-var.html#obj
+sub vcl_deliver {
+  # What is the remaining TTL for this object?
+  set resp.http.x-ttl = obj.ttl;
+  # What is the max object staleness permitted?
+  set resp.http.x-grace = obj.grace;
+
+  # Did the response come from Varnish or from the backend?
+  if (obj.hits > 0) {
+    set resp.http.x-cache = "HIT";
+  } else {
+    set resp.http.x-cache = "MISS";
+  }
+
+  # Is this object stale?
+  if (obj.ttl < std.duration(integer=0)) {
+    set resp.http.x-cache = "STALE";
+  }
+
+  # How many times has this response been served from Varnish?
+  set resp.http.x-cache-hits = obj.hits;
```

Comment:
It's perfectly fine to use the `x-` prefix, but [this practice was deprecated in 2012](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers).

Reply:
That is good to know!

Having looked at [IANA Hypertext Transfer Protocol (HTTP) Field Name Registry](https://www.iana.org/assignments/http-fields/http-fields.xhtml), the header which seems closest to what we want is [Cache-Status](https://www.rfc-editor.org/rfc/rfc9211.html). I have noticed that this does not mention anything about `hits` or `grace`, so I am wondering if the following looks good to you:

```console
Cache-Status: Shield; hit; hits=101; ttl=53.00; grace=86400.000, Edge; stale; hits=2; ttl=-123.00; grace=3600.000
```

This is how we are currently capturing this information - we don't yet have `Shield`, only `Edge`:

```console
x-cache: HIT
x-cache-hits: 1
x-ttl: 55.312
x-grace: 86400.000
```